worker/uniter/relation/resolver_test.go

```diff
@@ -142,23 +140,26 @@ func (s *relationResolverSuite) setupRelations(c *gc.C) relation.RelationStateTr
 
 	var numCalls int32
 	unitEntity := params.Entities{Entities: []params.Entity{{Tag: "unit-wordpress-0"}}}
+	unitStateResults := params.UnitStateResults{Results: []params.UnitStateResult{{}}}
 	apiCaller := mockAPICaller(c, &numCalls,
 		uniterAPICall("Refresh", unitEntity, params.UnitRefreshResults{Results: []params.UnitRefreshResult{{Life: life.Alive, Resolved: params.ResolvedNone}}}, nil),
 		uniterAPICall("GetPrincipal", unitEntity, params.StringBoolResults{Results: []params.StringBoolResult{{Result: "", Ok: false}}}, nil),
 		uniterAPICall("RelationsStatus", unitEntity, params.RelationUnitStatusResults{Results: []params.RelationUnitStatusResult{{RelationResults: []params.RelationUnitStatus{}}}}, nil),
+		uniterAPICall("State", unitEntity, unitStateResults, nil),
 	)
 	st := uniter.NewState(apiCaller, unitTag)
+	u, err := st.Unit(unitTag)
+	c.Assert(err, jc.ErrorIsNil)
 	r, err := relation.NewRelationStateTracker(
 		relation.RelationStateTrackerConfig{
 			State:                st,
-			UnitTag:              unitTag,
-			CharmDir:             s.stateDir,
-			RelationsDir:         s.relationsDir,
+			Unit:                 u,
+			CharmDir:             s.charmDir,
 			NewLeadershipContext: s.leadershipContextFunc,
 			Abort:                abort,
 		})
 	c.Assert(err, jc.ErrorIsNil)
-	assertNumCalls(c, &numCalls, 3)
+	assertNumCalls(c, &numCalls, 4)
```

Comment:
I had to resort to the same whack-a-mole game when I did the previous refactoring work here. We should really really try to rewrite these tests at some point to use mocks in the correct way...